tests/Fixtures/Integration/misc/PHP8_2.test

```diff
@@ -0,0 +1,179 @@
+--TEST--
+PHP 8.2 test.
+--RULESET--
+{
+    "@PhpCsFixer": true,
+    "@PHP82Migration": true
+}
+--REQUIREMENTS--
+{"php": 80200}
+--EXPECT--
+<?php
+
+// https://wiki.php.net/rfc/readonly_classes
+readonly final class Foo
+{
+    public string $c;
+    public string $prop;
+    public string $b;
+}
+
+// https://wiki.php.net/rfc/null-false-standalone-types
+class FalseNull
+{
+    public function __construct(
+        private false $falsy = false,
+        private null $nully = null
+    ) {
+    }
+
+    public function falsy(): false
+    {
+        return $this->falsy;
+    }
+
+    public function nully(): null
+    {
+        return $this->nully;
+    }
+}
+
+// https://wiki.php.net/rfc/true-type
+class TrueType
+{
+    public function __construct(
+        private true $havingABud = true
+    ) {
+    }
+
+    public function havingABud(): true
+    {
+        return $this->havingABud;
+    }
+}
+
+// https://wiki.php.net/rfc/dnf_types
+interface DnfA
+{
+    public function foo(): void;
+
+    public function bar(): void;
+}
+interface DnfB
+{
+    public function foo(): void;
+}
+interface DnfC
+{
+    public function bar(): void;
+}
+
+class DnfTypes
+{
+    public function __construct(
+        private DnfA|(DnfB&DnfC) $notSure
+    ) {
+    }
+
+    public function notSure(): DnfA|(DnfB&DnfC)
+    {
+        return $this->notSure;
+    }
+}
+
+// https://wiki.php.net/rfc/constants_in_traits
+trait WithConstants
+{
+    public const ONE = 'one';
+    protected const TWO = 'two';
+    private const THREE = 'three';
+}
+
+// https://wiki.php.net/rfc/fetch_property_in_const_expressions
+enum E: string
+{
+    case Foo = 'foo';
+}
+
+$eFoo = E::Foo->value;
```

Comment:
I'm surprised it did not fail 🤔 Maybe more complex usage should be added here, RFC contains more examples.

Reply:
Might be good to look into, however the syntax itself doesn't seem that new, it is just now OK to use it with enums. The fixer doesn't execute or validates PHP code beyond if it lints (into tokens), as such not all new PHP features will cause failures.

Reply:
I've added more fixtures and it works in general, but there's a [doubt about casing](https://github.com/FriendsOfPHP/PHP-CS-Fixer/issues/6613#issuecomment-1237508023), let's move discussion there.